Impossible 'body' field required in GET

**Description**

I'm seeing an issue where generated typescript types result in some impossible types:

```
Type error: Argument of type '{ params: { path: { id: string; }; }; }' is not assignable to parameter of type 'FetchOptions<{ parameters: { path: { id: string; }; }; responses: ...snipped
  Property 'body' is missing in type '{ params: { path: { id: string; }; }; }' but required in type '{ body: never; }'.

  10 |
  11 | async function exampleGet() {
> 12 |   const { data, error } = await get(`/v1/endpoint`, {
     |                                                                             ^
  13 |     params: { path: { id: "todo" } },
  14 |   });
  15 | }
```

I am able to work around it by adding a field `body: "" as never`, which seems to be forcing TS into the right path.  `body: undefined` or `body: null` does not work either.

**Reproduction**

I used the latest versions of openapi-fetch, openapi-typescript, and an internal OpenAPI spec with only GETs in it.

Something relevant in the generated code seems to be:
```
export interface components {
...
  responses: never;
  parameters: never;
  requestBodies: never;  // <--
  headers: never;
  pathItems: never;
}
```

**Expected result**

I can work out the types in this scenario more easily.

**Checklist**

- [X] I’m willing to open a PR (see [CONTRIBUTING.md](https://github.com/drwpow/openapi-fetch/blob/main/CONTRIBUTING.md))

@drwpow I'm also having this exact issue on (every) `GET` requests on `openapi-fetch@0.6.0` (types generated with `openapi-typescript@6.2.8`)

```
error TS2345: Argument of type '{ params: { query: { connectionId: number; }; path: { provider: string; operationId: string; }; }; }' is not assignable to parameter of type 'FetchOptions<{ parameters: { query: { connectionId: number; }; path: { provider: string; operationId: string; }; }; responses: { 200: { content: { "application/json": { message?: string; operationsLog?: { jobSummary?: { executionSeconds?: string; executeOperations?: string[]; okCount?: number; errorCount?: number; }...'.
  Property 'body' is missing in type '{ params: { query: { connectionId: number; }; path: { provider: string; operationId: string; }; }; }' but required in type '{ body: never; }'.
```

`body: undefined as never` does work. Mind you, vscode does not complain, workaround or not, but `tsc` does 🤷 

I can't share a full spec atm, but the types generated below map to the error above. If I had to bet, this seems like a generic "applies to all GET operations"-kinda issue

```ts
export interface paths {
  "/api/1.0/integrations/{provider}/{operationId}?connectionId={connectionId}": {
    get: operations["providerOperationId"];
  };
}

// ...

export interface operations {
  providerOperationId: {
    parameters: {
      query: {
        connectionId: components["parameters"]["connectionIdParam"];
      };
      path: {
        provider: components["parameters"]["providerParam"];
        operationId: components["parameters"]["operationIdParam"];
      };
    };
    responses: {
      200: {
        content: {
          "application/json": components["schemas"]["OperationOutcome"];
        };
      };
      401: components["responses"]["Unauthorized"];
      500: components["responses"]["Error"];
    };
  };
}
```

(PS: Thank you for making this tool! Bug aside, what I've seen so far is awesome! 👏 )

--- a/.changeset/cold-pets-sit.md
+++ b/.changeset/cold-pets-sit.md
@@ -1,0 +1,5 @@
+---
+"openapi-fetch": patch
+---
+
+Fix impossible body typing

--- a/packages/openapi-fetch/src/index.test.ts
+++ b/packages/openapi-fetch/src/index.test.ts
@@ -235,7 +235,7 @@ describe("client", () => {
 
         // expect error on missing `body`
         // @ts-expect-error
-        await client.GET("/blogposts", {});
+        await client.PUT("/blogposts", {});
 
         // expect error on missing fields
         // @ts-expect-error
@@ -401,9 +401,8 @@ describe("client", () => {
     it("returns empty object on 204", async () => {
       const client = createClient<paths>();
       mockFetchOnce({ status: 204, body: "" });
-      const { data, error, response } = await client.PUT("/tag/{name}", {
+      const { data, error, response } = await client.DELETE("/tag/{name}", {
         params: { path: { name: "New Tag" } },
-        body: { description: "This is a new tag" },
       });
 
       // assert correct data was returned

--- a/packages/openapi-fetch/src/index.ts
+++ b/packages/openapi-fetch/src/index.ts
@@ -51,7 +51,7 @@ export type Params<T> = T extends { parameters: any } ? { params: NonNullable<T[
 export type RequestBodyObj<T> = T extends { requestBody?: any } ? T["requestBody"] : never;
 export type RequestBodyContent<T> = undefined extends RequestBodyObj<T> ? FilterKeys<NonNullable<RequestBodyObj<T>>, "content"> | undefined : FilterKeys<RequestBodyObj<T>, "content">;
 export type RequestBodyMedia<T> = FilterKeys<RequestBodyContent<T>, MediaType> extends never ? FilterKeys<NonNullable<RequestBodyContent<T>>, MediaType> | undefined : FilterKeys<RequestBodyContent<T>, MediaType>;
-export type RequestBody<T> = undefined extends RequestBodyMedia<T> ? { body?: RequestBodyMedia<T> } : { body: RequestBodyMedia<T> };
+export type RequestBody<T> = RequestBodyMedia<T> extends never ? { body?: never } : undefined extends RequestBodyMedia<T> ? { body?: RequestBodyMedia<T> } : { body: RequestBodyMedia<T> };
 export type QuerySerializer<T> = (query: T extends { parameters: any } ? NonNullable<T["parameters"]["query"]> : Record<string, unknown>) => string;
 export type BodySerializer<T> = (body: RequestBodyMedia<T>) => any;
 export type RequestOptions<T> = Params<T> &

--- a/packages/openapi-fetch/test/v1.d.ts
+++ b/packages/openapi-fetch/test/v1.d.ts
@@ -176,6 +176,18 @@ export interface paths {
         500: components["responses"]["Error"];
       };
     };
+    delete: {
+      parameters: {
+        path: {
+          name: string;
+        };
+      };
+      responses: {
+        /** @description No Content */
+        204: never;
+        500: components["responses"]["Error"];
+      };
+    };
     parameters: {
       path: {
         name: string;

--- a/packages/openapi-fetch/test/v1.yaml
+++ b/packages/openapi-fetch/test/v1.yaml
@@ -190,6 +190,12 @@ paths:
           $ref: '#/components/responses/CreateTag'
         500:
           $ref: '#/components/responses/Error'
+    delete:
+      responses:
+        204:
+          description: No Content
+        500:
+          $ref: '#/components/responses/Error'
   /default-as-error:
     get:
       responses:

